### PR TITLE
[patch] fix duplicated search result

### PR DESCRIPTION
**Pull request by kpango**
Signed-off-by: kpango <i.can.feel.gravity@gmail.com>

### Description:
fix duplicated search result

### Environment:

- Go Version: 1.15.2
- Docker Version: 19.03.8
- Kubernetes Version: 1.18.2
- NGT Version: 1.12.1

### Types of changes:

- [x] Bug fix [type/bug]
- [ ] New feature [type/feature]
- [ ] Add tests [type/test]
- [ ] Security related changes [type/security]
- [ ] Add documents [type/documentation]
- [ ] Refactoring [type/refactoring]
- [ ] Update dependencies [type/dependency]
- [ ] Update benchmarks and performances [type/bench]
- [ ] Update CI [type/ci]

### Changes to Core Features:

- [ ] Have you added an explanation of what your changes do and why you'd like us to include them?
- [ ] Have you written new tests for your core changes, as applicable?
- [ ] Have you successfully ran tests with your changes locally?

### Checklist:

- [ ] I have read the [CONTRIBUTING](https://github.com/vdaas/vald/blob/master/CONTRIBUTING.md) document.
- [ ] I have checked open [Pull Requests](https://github.com/vdaas/vald/pulls) for the similar feature or fixes?
- [ ] I have added tests and benchmarks to cover my changes.
- [ ] I have ensured all new and existing tests passed.
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

File: pkg/gateway/vald/handler/grpc/handler.go [hunk 1]
```diff
@@ -161,15 +161,13 @@ func (s *server) search(ctx context.Context, cfg *payload.Search_Config,
 					return nil
 				}
 				id := dist.GetId()
-				mu.RLock()
+				mu.Lock()
```
Comment (rinx) [inline pkg/gateway/vald/handler/grpc/handler.go:164]: i'm not sure but how about to use sync.Map as `visited`?

Comment (kpango) [reply pkg/gateway/vald/handler/grpc/handler.go:164]: we already tried sync.Map implementation, at https://github.com/vdaas/vald/pull/729/files#diff-320187c6fd3d930cdcd5999eb7192ed5R150 and https://github.com/vdaas/vald/pull/729/files#diff-320187c6fd3d930cdcd5999eb7192ed5R172-R175
if we change this logic to sync.Map we have to re-check accuracy.

Comment (kpango) [reply pkg/gateway/vald/handler/grpc/handler.go:164]: I think if we change mutex to lockfree we should make another PR, this PR is for current bug fix.

Comment (rinx) [reply pkg/gateway/vald/handler/grpc/handler.go:164]: aha, i see.

Comment (kpango) [reply pkg/gateway/vald/handler/grpc/handler.go:164]: But I wanted to use sync.Map for this block, I'll make proposal PR for it.

**Pull request merged**